desktop/cmp/panel/impl/PanelHeader.ts

```diff
@@ -42,11 +44,11 @@ export const panelHeader = hoistCmp.factory({
             compactCls = compact ? 'xh-panel-header--compact' : null;
 
         // 1) Classic "top" title bar
-        if (!collapsed || vertical || isModal) {
+        if ((!collapsed || vertical || isModal) && (title || icon || headerItems)) {
```

Comment:
What is this change doing here?  I don't think we need it - think we want to be checking off of `displayedTitle/Icon`, and the change to the early-out on line 32 above should take care of the case where there aren't any of these.

Although note that we deliberately don't render `headerItems` when collapsed - see below line 60/62. I'm just reading here in Github, but it seems to me that we could setup `displayedHeaderItems` above to be an empty array if `collapsed && !isModal` - then we really have all of our collapsed-sensitive state consistently available and ready to check.

Also - sorry just looking a bit more closely - we should be careful that the early-out above isn't violating the expectations/requirements of the modal support stuff - it's planning to put a button in the header.  

I think this all needs to be looked at a bit more carefully

Reply:
Now that I'm looking at it more closely, I'm not sure why we are showing the modal button in collapsed headers - in general but also specifically collapsed headers of top/bottom panels but not left/right panels. Why would someone want to pop out a collapsed panel? If it is something we want to continue to offer, shouldn't we do the same for all panels regardless of side?

Reply:
I think the aim there was to show a collapsed top/bottom header the same as an expanded one, with the same header tools and / or modal button. We can be reasonably sure that we have the same amount of space to work with and didn't want things to jump about. You can think of it as "the **content** goes away, and the existing header moves down".

Collapsing left / right is a different beast, as it changes the dimensions and orientation of the header. We can't reliably expect the available space to be equivalent - typically (as in the Portfolio example app) we'll have a lot less space to work with. You can think of it as "the **panel** goes away, and is replaced by a minimal affordance". Rotated text is harder to read, and rotated header items / buttons are just not particularly usable. There are also technical styling reasons for not attempting to rotate header items.

At least that's how I think about it 🤷‍♂️

Reply:
I like your way of thinking about it tom

Reply:
In new PR #3480 I went with consistently showing the modalToggle for both LR and TB collapsed panels - that seemed like a pretty clear choice to me, as long as we're OK with both buttons migrating to the bottom of the vertical header.

I still left out app-specific headerTools for collapsed panels, even TB-collapsed ones with the horizontal header. My thinking was that it's very likely that headerTools are only there to control something about the content of the panel, and it seemed potentially dangerous (or at least weird) to show them when you can't see that content.  

If we hit a specific use case where we do want them, we could consider adding a flag to control at that point - `showHeaderToolsWhenCollapsed` or something crazy like that :)